bin/extract_galaxy_tools.py

```diff
@@ -135,28 +135,23 @@ def check_categories(ts_categories, ts_cat):
     :param ts_cat: list of ToolShed categories to keep in the extraction
     """
     if ts_categories is not None and len(ts_cat) > 0:
+        ts_cats = ts_categories.split(", ")
         to_keep = False
-        for cat in ts_categories:
+        for cat in ts_cats:
             if cat in ts_cat:
                 to_keep = True
         return to_keep
     return True
 
 
-def get_tool_metadata(tool, repo, ts_cat, excluded_tools, keep_tools):
+def get_tool_metadata(tool: ContentFile, repo: Repository) -> :
```

Comment:
Sorry, forgot to finish this suggestion:
```suggestion
def get_tool_metadata(tool: ContentFile, repo: Repository) -> Optional[Dict[str, Any]]:
```

and needs also this change at the top:

```diff
diff --git a/bin/extract_galaxy_tools.py b/bin/extract_galaxy_tools.py
index 11cbd2d..9f32804 100644
--- a/bin/extract_galaxy_tools.py
+++ b/bin/extract_galaxy_tools.py
@@ -5,7 +5,12 @@ import base64
 import time
 import xml.etree.ElementTree as et
 from pathlib import Path
-from typing import List
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+)
 
 import pandas as pd
 import requests
```